lib/puppet/provider/rabbitmq_user_permissions/rabbitmqctl.rb

```diff
@@ -1,28 +1,27 @@
 require File.expand_path(File.join(File.dirname(__FILE__), '..', 'rabbitmqctl'))
-Puppet::Type.type(:rabbitmq_user_permissions).provide(:rabbitmqctl, :parent => Puppet::Provider::Rabbitmqctl) do
-
+Puppet::Type.type(:rabbitmq_user_permissions).provide(:rabbitmqctl, parent: Puppet::Provider::Rabbitmqctl) do
   if Puppet::PUPPETVERSION.to_f < 3
-    commands :rabbitmqctl => 'rabbitmqctl'
+    commands rabbitmqctl: 'rabbitmqctl'
   else
-     has_command(:rabbitmqctl, 'rabbitmqctl') do
-       environment :HOME => "/tmp"
-     end
+    has_command(:rabbitmqctl, 'rabbitmqctl') do
+      environment HOME: '/tmp'
+    end
   end
 
-  defaultfor :feature=> :posix
+  defaultfor :feature => :posix
 
   # cache users permissions
   def self.users(name, vhost)
     @users = {} unless @users
     unless @users[name]
       @users[name] = {}
-      self.run_with_retries {
+      run_with_retries do
         rabbitmqctl('-q', 'list_user_permissions', name)
-      }.split(/\n/).each do |line|
-        line = self::strip_backslashes(line)
-        if line =~ /^(\S+)\s+(\S*)\s+(\S*)\s+(\S*)$/
-          @users[name][$1] =
-            {:configure => $2, :read => $4, :write => $3}
+      end.split(%r{\n}).each do |line|
+        line = .strip_backslashes(line)
```

Comment:
Is this right?
Tests are failing here due to syntax, not rubocop.

Reply:
@alexjfisher yeah - I saw this last time too, for some reason, `rubocop -a` changed `self.strip_backslashes` to `.strip_backslashes` in that case. I added a separate commit (not squashed for this case, since it's a manual fix) that should resolve that particular issue. Let's see if the tests are at least somewhat happier after that.

There's one other issue that relates to spacing in the tests, I think I'll need to switch it to a heredoc to match it.